python/foglamp/services/south/server.py

```diff
@@ -210,19 +181,40 @@ def run(self):
         for signal_name in (signal.SIGINT, signal.SIGTERM):
             loop.add_signal_handler(
                 signal_name,
-                lambda: asyncio.ensure_future(self._stop(loop)))
+                lambda: asyncio.ensure_future(self.stop()))
 
         asyncio.ensure_future(self._start(loop))
         loop.run_forever()
 
-    async def shutdown(self, request):
+    async def _stop(self):
+        if self._plugin is not None:
+            try:
+                self._plugin.plugin_shutdown(self._plugin_handle)
+            except Exception:
+                _LOGGER.exception("Unable to stop plugin '{}'".format(self._name))
+            finally:
+                self._plugin = None
+                self._plugin_handle = None
+
+        try:
+            await Ingest.stop()
+        except Exception:
+            _LOGGER.exception('Unable to stop the Ingest server')
+            return
+
+        # Stop all pending asyncio tasks
+        for task in asyncio.Task.all_tasks():
+            task.cancel()
+
+    async def stop(self, request=None):
```

Comment:
shutdown with request arg; Again interesting will be when to kill service' (managment port) aiohttp instance and return response to microservice management client api